packages/bcer-api/app/src/location/entities/location.entity.ts

```diff
@@ -95,34 +95,38 @@ export class LocationEntity {
   @OneToOne(
     () => NoiEntity,
     (noi: NoiEntity) => noi.location,
-    { nullable: true },
+    { nullable: true, onDelete: 'CASCADE' },
   )
   @JoinColumn()
   noi: NoiEntity;
 
   @ManyToMany(
     () => ProductEntity,
     (product: ProductEntity) => product.locations,
+    { onDelete: 'CASCADE' }
   )
   @JoinTable()
   products: ProductEntity[];
 
   @OneToMany(
     () => ProductSoldEntity,
     (productSold: ProductSoldEntity) => productSold.location,
+    { onDelete: 'CASCADE' }
```

Comment:
not exactly sure but it looks like the onDelete needs to be only on the @ManyToOne side of relationship.